perl/cpan/private/carton.bzl

```diff
@@ -0,0 +1,199 @@
+"""Bazel tools for interfacing with Carton"""
+
+_HUB_BUILD_FILE = """\
+load("@rules_perl//perl:perl_library.bzl", "perl_library")
+
+package(default_visibility = ["//visibility:public"])
+
+DEPENDENCIES = {dependencies}
+
+perl_library(
+    name = "{name}",
+    deps = [
+        "@{name}__" + dep
+        for dep in DEPENDENCIES
+    ],
+    includes = [],
+)
+
+[
+    alias(
+        name = dep,
+        actual = "@{name}__" + dep,
+    )
+    for dep in DEPENDENCIES
+]
+"""
+
+def _perl_cpan_hub_impl(repository_ctx):
+    repository_ctx.file("WORKSPACE.bazel", """workspace(name = "{}")""".format(
+        repository_ctx.name,
+    ))
```

Comment:
Curious, what's the effect of explicitly setting the workspace name? An empty file works well in my experience

Reply:
An empty file will work, this is mostly out of habit to improve debug-ability I've benefited from in the past.